openstudiocore/src/model/SurfacePropertyConvectionCoefficients.cpp

```diff
@@ -0,0 +1,399 @@
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **********************************************************************************************************************/
+
+#include "SurfacePropertyConvectionCoefficients.hpp"
+#include "SurfacePropertyConvectionCoefficients_Impl.hpp"
+
+#include "Model.hpp"
+#include "Model_Impl.hpp"
+#include "Surface.hpp"
+#include "Surface_Impl.hpp"
+#include "SubSurface.hpp"
+#include "SubSurface_Impl.hpp"
+#include "InternalMass.hpp"
+#include "InternalMass_Impl.hpp"
+#include "Schedule.hpp"
+#include "Schedule_Impl.hpp"
+
+#include <utilities/idd/IddFactory.hxx>
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_SurfaceProperty_ConvectionCoefficients_FieldEnums.hxx>
+
+#include "../utilities/units/Unit.hpp"
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  SurfacePropertyConvectionCoefficients_Impl::SurfacePropertyConvectionCoefficients_Impl(const IdfObject& idfObject,
+                                                                                         Model_Impl* model,
+                                                                                         bool keepHandle)
+    : ModelObject_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == SurfacePropertyConvectionCoefficients::iddObjectType());
+  }
+
+  SurfacePropertyConvectionCoefficients_Impl::SurfacePropertyConvectionCoefficients_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                                                                         Model_Impl* model,
+                                                                                         bool keepHandle)
+    : ModelObject_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == SurfacePropertyConvectionCoefficients::iddObjectType());
+  }
+
+  SurfacePropertyConvectionCoefficients_Impl::SurfacePropertyConvectionCoefficients_Impl(const SurfacePropertyConvectionCoefficients_Impl& other,
+                                                                                         Model_Impl* model,
+                                                                                         bool keepHandle)
+    : ModelObject_Impl(other,model,keepHandle)
+  {}
+
+  const std::vector<std::string>& SurfacePropertyConvectionCoefficients_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty()){
+    }
+    return result;
+  }
+
+  IddObjectType SurfacePropertyConvectionCoefficients_Impl::iddObjectType() const {
+    return SurfacePropertyConvectionCoefficients::iddObjectType();
+  }
+
+  bool SurfacePropertyConvectionCoefficients_Impl::setSurface(const ModelObject &surface) {
+    return setPointer(OS_SurfaceProperty_ConvectionCoefficientsFields::SurfaceName, surface.handle());
```

Comment:
If we did these things the only way to get two of these objects pointing at the same surface is to clone it.  We can make clone blank out the surface field.